File: api/json.go
package handler

import (
	"encoding/json"
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"time"
)

func Jsonparse(w http.ResponseWriter, r *http.Request) {
	var parsed map[string]interface{}
	res, err := http.Get("https://pokeapi.co/api/v2/pokemon?limit=100000&offset=0")
	if err != nil {
		log.Fatalln(err)
	}

	body, err := ioutil.ReadAll(res.Body)
	if err != nil {
		log.Fatalln(err)
	}
	jsonStr := string(body)

	start := time.Now()
	json.Unmarshal([]byte(jsonStr), &parsed)
	duration := time.Since(start)

	// fmt.Fprintf(w, "len=%d cap=%d %v\n", len(s), cap(s), s)
	fmt.Fprintf(w, "%f ms", float64(duration.Nanoseconds())/1000000)
}

File: api/rust/json.rs
use serde_json::{Map, Value};
use std::time::Instant;
use vercel_runtime::{run, Body, Error, Request, Response, StatusCode};

#[tokio::main]
async fn main() -> Result<(), Error> {
    run(handler).await
}

pub async fn handler(_req: Request) -> Result<Response<Body>, Error> {
    let response =
        reqwest::blocking::get("https://pokeapi.co/api/v2/pokemon?limit=100000&offset=0").unwrap();
    let json = response.text().unwrap();

    let start = Instant::now();
    let parsed: Value = serde_json::from_str(&json)?;
    let _obj: Map<String, Value> = parsed.as_object().unwrap().clone();
    let duration = start.elapsed();

    let time = format!("{:?} ms", duration.as_secs_f64() * 1000.0);
    Ok(Response::builder()
        .status(StatusCode::OK)
        .body(time.into())?)
}
